nir/src/main/scala/scala/scalanative/nir/serialization/BinarySerializer.scala

```diff
@@ -387,6 +387,16 @@ final class BinarySerializer(buffer: ByteBuffer) {
     case Type.Module(n) => putInt(T.ModuleType); putGlobal(n)
   }
 
+  private def putArgs(args: Seq[Arg]): Unit = putSeq(args)(putArg)
+  private def putArg(arg: Arg): Unit = {
+    putType(arg.ty)
+    putSeq(arg.attrs.toSeq)(putArgAttr)
```

Comment:
Once `Arg` is a single-level wrapper, you'll be able to use `putOpt` to serialize its fields.